src/cli/commands/pushMarketing.ts

```diff
@@ -0,0 +1,199 @@
+import {
+  asArray,
+  asEither,
+  asMaybe,
+  asNull,
+  asNumber,
+  asString
+} from 'cleaners'
+import { Command, Option } from 'clipanion'
+import { viewToStream } from 'edge-server-tools'
+
+import { asCouchDevice } from '../../db/couchDevices'
+import { makePushSender } from '../../util/pushSender'
+import { ServerContext } from '../cliTools'
+
+// Firebase only allows a certain number of tokens per invocation of the API
+const MAX_FIREBASE_BATCH_TOKEN_COUNT = 500
+
+// ANSI Terminal Escape Sequences:
+const ESC = `\x1B[`
+const SAVE_POSITION = `${ESC}s`
+const RESOTRE_POSITION = `${ESC}u`
+const CLEAR_LINE = `${ESC}2K`
+const CLEAR_AND_RESORE = `${CLEAR_LINE}${RESOTRE_POSITION}`
```

Comment:
If you look in the loginServer, we have a "heatbeat" thing that prints out progress as we walk over the database. If you could bundle this up into a similar API, it would be much more usable, and maybe we will steal it into the login server too. I really appreciate the direct ANSI approach, as opposed to pulling in a dependency.